lib/HTFeed/Stage/ImageRemediate.pm

```diff
@@ -412,21 +425,31 @@ sub repair_tiff_exiftool {
 
     # whines if infile is same as outfile
     my @file_params = ($infile);
-    push( @file_params, $outfile ) if ( $outfile ne $infile );
+    push(@file_params, $outfile) if ($outfile ne $infile);
 
-    if ( !$exifTool->WriteInfo(@file_params) ) {
-        croak(  "Couldn't remediate $infile: "
-              . $exifTool->GetValue('Error')
-              . "\n" );
+    my $write_return = $exifTool->WriteInfo(@file_params);
+    if (!$write_return) {
+        croak(
+            "Couldn't remediate $infile: ". $exifTool->GetValue('Error') . "\n"
+        );
         return 0;
     }
+    my $end_time = $self->{job_metrics}->time;
+    my $delta_time = $end_time - $start_time;
+    $self->{job_metrics}->inc("ingest_imageremediate_images");
```

Comment:
We should add some labels here to differentiate between exiftool, imagemagick, tiff repair, etc. It might also make sense to extract this as a subroutine since we are doing something with a very similar shape in multiple places in `ImageRemediate`.

Reply:
@aelkiss maybe it is a shortcoming of my imagination, but any extracted subroutine I picture is going to be about as complicated to call as these existing lines. Each time we make a (set of) measurement(s) of an operation we need to determine 1) when the operation started, 2) ended, 3) which file sizes meaningfully changed, 4) which labels to use, 5) which metric to increment, 6) ...?

I can imagine writing the time measurement as a callback:

```
my $timer = $job_metrics->start_timer("ingest_pack_items", $labels);
... # time passes, work happens
$timer->stop_and_record(); # writes delta time since get_timer to prometheus
```
... but how would I do that for file & dir sizes? I guess I could pass in the paths for inputs and outputs. But at this point I'm using about as many lines of code each time setting up callbacks as I'm currently using, and I'm starting to lose the point.

Do tell me if you have strong feelings about this, or if I'm misunderstanding your feedback.

Reply:
I think fine to leave as-is for now. I'll take a look and see if I can propose something at a future point.